test/watch/test/file-watching.test.ts

```diff
@@ -40,10 +40,6 @@ afterEach(() => {
   cleanups.splice(0).forEach(cleanup => cleanup())
 })
 
-// TODO: Fix flakiness and enable on CI
-if (process.env.GITHUB_ACTIONS)
-  test.only('skip tests on CI', () => {})
-
 test('editing source file triggers re-run', async () => {
```

Comment:
I think the flakiness in files that rely on watch mode might be due to the watcher incorrectly emitting ready event too soon. You can try incorporating the utility I wrote for Vite tests from here: https://github.com/sheremet-va/vite/blob/4df5ac29087f448a40f80707dec78d197e74c6d2/playground/hmr-ssr/__tests__/hmr.spec.ts#L960

We already check for a ready event with an env variable for tests:

https://github.com/vitest-dev/vitest/blob/8877e22a090c5a5ccfd9206122649181ba9c44f9/packages/vite-node/src/hmr/emitter.ts#L54

Reply:
Thank you for your advice!
I will check it out!